src/datetime.js

```diff
@@ -1548,12 +1548,85 @@ export default class DateTime {
    * @example DateTime.now().toISO({ format: 'basic' }) //=> '20170422T204705.335-0400'
    * @return {string}
    */
-  toISO(opts = {}) {
+  toISO({
+    format = "extended",
+    suppressSeconds = false,
+    suppressMilliseconds = false,
+    includeOffset = true,
+  } = {}) {
+    function simplifiedPadStart(positiveInteger, n = 2) {
+      return ("0".repeat(n) + positiveInteger).slice(-n);
+    }
+
     if (!this.isValid) {
       return null;
     }
 
-    return `${this.toISODate(opts)}T${this.toISOTime(opts)}`;
+    const o = this;
+    if (format === "extended") {
+      let c = simplifiedPadStart(o.c.year, 4);
+      c += "-";
+      c += simplifiedPadStart(o.c.month);
+      c += "-";
+      c += simplifiedPadStart(o.c.day);
+      c += "T";
+      c += simplifiedPadStart(o.c.hour);
+      c += ":";
+      c += simplifiedPadStart(o.c.minute);
+      if (o.c.second !== 0 || !suppressSeconds) {
+        c += ":";
+        c += simplifiedPadStart(o.c.second);
+        if (o.c.millisecond !== 0 || !suppressMilliseconds) {
+          c += ".";
+          c += simplifiedPadStart(o.c.millisecond, 3);
+        }
+      }
+
+      if (includeOffset) {
+        if (o.o === 0) {
```

Comment:
This isn't quite right. `Z` is only if it's UTC, and some peoples' zones are just at 00:00. The should get "00:00"

Reply:
Thanks ! Fixed